doc/index.rst

```diff
@@ -156,7 +156,7 @@ You can configure the behavior in `CI config <#terminology>`_ in your client rep
 
 Required environment variables:
 
-* `ROS_DISTRO`: Version of ROS in all lower case. E.g.: `indigo` / `jade`
+* `ROS_DISTRO`: Version of ROS in all lower case. E.g.: `indigo` / `jade`. If is is set in the custom Docker (base) image, it might be omitted in the script call.
```

Comment:
If the user specifies `DOCKER_BASE_IMAGE` or `DOCKER_IMAGE` and any of these images have the ROS_DISTRO variable set, then this setting can be omitted (=left out of the config).

In case of `DOCKER_BASE_IMAGE` `ROS_DISRO` can be overridden (e.g. `jade` images might be used for 
`indgo` as well, with limited reuse). In case of `DOCKER_IMAGE`, setting a different `ROS_DISTRO` will fail due to missing files/packages etc.

> "Omitted" sounds as if ROS_DISTRO would be simply removed and remain unset.

I it safe to remove it in almost all cases.

> Also, should we recommend (maybe at its description) to not specify ROS_DISTRO when the user knows ROS_DISTRO is already defined in the Docker image passed so that it'll automatically be set?

I don't know if I would recommend it by now.
I just wanted to express that there is at least the possibility to omit it in certain cases.
As I explained above it does not do any harm if the `ROS_DISTRO` is set to the same value twice.

Reply:
> If the user specifies DOCKER_BASE_IMAGE or DOCKER_IMAGE and any of these images have the ROS_DISTRO variable set, then this setting can be omitted (=left out of the config).

I see. So, the `ROS_DISTRO` defined in the image is prioritized, and the one in the config will be ignored.

> In case of DOCKER_BASE_IMAGE ROS_DISRO can be overridden (e.g. jade images might be used for indgo as well, with limited reuse). In case of DOCKER_IMAGE, setting a different ROS_DISTRO will fail due to missing files/packages etc.

Interesting...I'd say that may be unusual case so that we don't have to document.